### ISSUE TITLE
Adam Optimizer not working in migration of custom model

### ISSUE THREAD

**Issue by zaza-ipynb** [tensorflow/tensorflow#43968]
I'm performing a migration to my custom model from TensorFlow 1.x to TensorFlow 2.x, so parts of my custom model (Class Model) are wrapped in a function that decorated with tf.function so before I use session.run with feed_dict, but now I use a function that decorated with tf.function, the problem is that the accuracy is small and the loss is big, so I implement Adam optimizer inside the decorated function with tf.function (def new_function), but it shows error creating variable on non-first call, however, it runs when I declare Adam optimizer in forecast function(def forecast) where I call the Model to predict, but it doesn't affect the accuracy that should be increasing and the loss should be minimized, it just remains the same like without using adam optimizer but slower. Is something missing? or how should be the syntax written in the code and where should it be placed?

note : Adam are implemented with 3 lines of code with comment in def forecast, see the difference of speed with or without Adam, and why the accuracy is not increasing and loss is also still big.

https://colab.research.google.com/drive/1VpahdrELXkMk29670yxLksOJMoR7uHKa
this is the colab pushed file

kindly need help for this problem

**Comment by zaza-ipynb** [comment]
Hey sir @ravikyram thanks for the respond, is it because it is still using lstmcell and multirnn from tensorflow.compat? bcs that is the only thing that shows, and i still dont understand what is the issue with adam optimizer in def forecast not giving any effect in the accuracy and loss of training. Thank You for the help, still waiting others 👍

**Comment by ymodak** [comment]
Apologies for the delay in response. This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there. The response chances highly increase if you can come up with minimal code repro.
Thanks!
